lib/rules/no-negated-condition.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @fileoverview Rule to disallow a negated condition
+ * @author Brandon Mills
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    return {
+        "IfStatement": function(node) {
+            if (!node.alternate || node.alternate.type === "IfStatement") {
+                return;
+            }
+            if ((node.test.type === "UnaryExpression" && node.test.operator === "!") ||
+                (node.test.type === "BinaryExpression" &&
+                (node.test.operator === "!=" || node.test.operator === "!=="))) {
+                context.report(node, "Unexpected negated condition.");
+            }
+        },
+        "ConditionalExpression": function(node) {
+            if (node.test.type === "UnaryExpression" && node.test.operator === "!") {
```

Comment:
You should probably also have the BinaryExpression test here, to align with IfStatement test handling.